$NetBSD: patch-data_CMakeLists.txt,v 1.2 2024/02/10 09:30:43 ryoon Exp $

* Fix runtime errors during build for pkgsrc.
  pkgsrc removes $ORIGIN from RPATH.

--- data/CMakeLists.txt.orig	2023-07-13 07:06:39.663232800 +0000
+++ data/CMakeLists.txt
@@ -19,7 +19,7 @@ add_custom_target(lm ALL DEPENDS "${LM_O
 add_custom_command(
   OUTPUT "${LM_PREDICT_OUTPUT}"
   DEPENDS "${LM_SRC}" lm LibIME::prediction
-  COMMAND LibIME::prediction "${LM_OUTPUT}" "${LM_SRC}" "${LM_PREDICT_OUTPUT}")
+  COMMAND "${CMAKE_COMMAND}" -E env "LD_LIBRARY_PATH=${PROJECT_BINARY_DIR}/src/libime/core" ${PROJECT_BINARY_DIR}/tools/libime_prediction "${LM_OUTPUT}" "${LM_SRC}" "${LM_PREDICT_OUTPUT}")
 add_custom_target(lm-predict ALL DEPENDS "${LM_PREDICT_OUTPUT}")
 
 install(FILES "${LM_OUTPUT}" RENAME zh_CN.lm DESTINATION "${LIBIME_INSTALL_LIBDATADIR}")
@@ -38,14 +38,14 @@ set(DICT_OUTPUT "${CMAKE_CURRENT_BINARY_
 add_custom_command(
   OUTPUT "${DICT_OUTPUT}"
   DEPENDS "${DICT_SRC}" LibIME::pinyindict
-  COMMAND LibIME::pinyindict "${DICT_SRC}" "${DICT_OUTPUT}")
+  COMMAND "${CMAKE_COMMAND}" -E env "LD_LIBRARY_PATH=${PROJECT_BINARY_DIR}/src/libime/pinyin:${PROJECT_BINARY_DIR}/src/libime/core" ${PROJECT_BINARY_DIR}/tools/libime_pinyindict "${DICT_SRC}" "${DICT_OUTPUT}")
 
 set(DICT_EXTB_SRC "${CMAKE_CURRENT_BINARY_DIR}/dict_extb.txt")
 set(DICT_EXTB_OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/extb.dict")
 add_custom_command(
   OUTPUT "${DICT_EXTB_OUTPUT}"
   DEPENDS "${DICT_EXTB_SRC}" LibIME::pinyindict
-  COMMAND LibIME::pinyindict "${DICT_EXTB_SRC}" "${DICT_EXTB_OUTPUT}")
+  COMMAND "${CMAKE_COMMAND}" -E env "LD_LIBRARY_PATH=${PROJECT_BINARY_DIR}/src/libime/pinyin:${PROJECT_BINARY_DIR}/src/libime/core" ${PROJECT_BINARY_DIR}/tools/libime_pinyindict "${DICT_EXTB_SRC}" "${DICT_EXTB_OUTPUT}")
 add_custom_target(dict ALL DEPENDS "${DICT_OUTPUT}" "${DICT_EXTB_OUTPUT}")
 install(FILES "${DICT_OUTPUT}" "${DICT_EXTB_OUTPUT}" DESTINATION "${LIBIME_INSTALL_PKGDATADIR}")
 
@@ -63,7 +63,7 @@ foreach(TABLE_TXT_FILE ${TABLE_TXT_FILES
   string(REPLACE .txt .main.dict TABLE_DICT_FILE ${TABLE_TXT_FILE})
   add_custom_command(OUTPUT ${TABLE_DICT_FILE}
                      DEPENDS ${TABLE_TXT_FILE} LibIME::tabledict
-                     COMMAND LibIME::tabledict ${TABLE_TXT_FILE} ${CMAKE_CURRENT_BINARY_DIR}/${TABLE_DICT_FILE})
+                     COMMAND "${CMAKE_COMMAND}" -E env "LD_LIBRARY_PATH=${PROJECT_BINARY_DIR}/src/libime/table:${PROJECT_BINARY_DIR}/src/libime/core" ${PROJECT_BINARY_DIR}/tools/libime_tabledict ${TABLE_TXT_FILE} ${CMAKE_CURRENT_BINARY_DIR}/${TABLE_DICT_FILE})
   list(APPEND TABLE_DICT_FILES ${CMAKE_CURRENT_BINARY_DIR}/${TABLE_DICT_FILE})
 endforeach()
 
